### PR TITLE
feat: add error handling in enrollment and onboardingStatus component

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2419](https://openedx.atlassian.net/browse/PROD-2419)

Screenshot 2021-07-09 at 4 43 07 PM

Screenshot 2021-07-09 at 5 49 22 PM

### CODE REVIEW

File: src/users/data/api.js [hunk 1]
```diff
@@ -33,10 +33,24 @@ export async function getEntitlements(username, page = 1) {
...
+      AppUrls.getEnrollmentsUrl(username),
+    );
+    return data;
+  } catch (error) {
+    return {
+      errors: [
+        {
+          code: null,
+          dismissible: true,
+          text: 'An unexpected error occurred. Please try refreshing the page.',
```
Comment (DawoudSheraz) [inline src/users/data/api.js]: maybe use error's own text instead of hardcoding the error text?

File: src/users/data/api.test.js [hunk 2]
```diff
@@ -83,6 +95,13 @@ describe('API', () => {
       const response = await api.getOnboardingStatus(data, testUsername);
       expect(response).toEqual({ ...expectedSuccessResponse, onboardingStatus: 'Error while fetching data' });
     });
+
+    it('Unexpected error fetching enrollments', async () => {
+      // mockAdapter.onGet(url).reply(() => throwError(500, ''));
```
Comment (DawoudSheraz) [inline src/users/data/api.test.js]: what's going on here?

Comment (Ali-D-Akbar) [reply src/users/data/api.test.js]: Removing this.

File: src/users/data/api.js (hunk 1 above)
Comment (Ali-D-Akbar) [reply src/users/data/api.js]: Okay.

File: src/users/enrollments/Enrollments.jsx [hunk 3]
```diff
@@ -57,7 +68,7 @@ export default function Enrollments({
   });
 
   const tableData = useMemo(() => {
-    if (enrollmentData === null || enrollmentData.length === 0) {
+    if (enrollmentData === null || enrollmentData.errors || enrollmentData.length === 0) {
```
Comment (DawoudSheraz) [inline src/users/enrollments/Enrollments.jsx]: where is enrollmentData.errors coming from?

**Review by DawoudSheraz — COMMENTED** [review]
added some comments

File: src/users/enrollments/Enrollments.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/enrollments/Enrollments.jsx]: from api when there's an error.

```
  try {
    const { data } = await getAuthenticatedHttpClient().get(
      AppUrls.getEnrollmentsUrl(username),
    );
    return data;
  } catch (error) {
    return {
      errors: [
        {
          code: null,
          dismissible: true,
          text: JSON.parse(error.customAttributes.httpErrorResponseData),
          type: 'danger',
          topic: 'enrollments',
        },
      ],
    };
  }
```

Comment (DawoudSheraz) [reply src/users/enrollments/Enrollments.jsx]: but the error response is added to alert and not to enrollmentData.

File: src/users/data/api.js [hunk 4]
```diff
@@ -33,10 +33,24 @@ export async function getEntitlements(username, page = 1) {
...
+      AppUrls.getEnrollmentsUrl(username),
+    );
+    return data;
+  } catch (error) {
+    return {
+      errors: [
+        {
+          code: null,
+          dismissible: true,
+          text: JSON.parse(error.customAttributes.httpErrorResponseData),
```
Comment (azanbinzahid) [inline src/users/data/api.js:47]: will it always have `customAttributes`? what-if it does not?

File: src/users/enrollments/Enrollments.jsx (hunk 3 above)
Comment (Ali-D-Akbar) [reply src/users/enrollments/Enrollments.jsx]: Oh, it is not needed here. I'll remove this.

File: src/users/data/api.js (hunk 4 above)
Comment (Ali-D-Akbar) [reply src/users/data/api.js:47]: I'm not so sure about this but as far as I know, the standard error object in react has `customAttributes` extracted from the backend. So we may always have a `customAttributes` in `error`

Comment (DawoudSheraz) [reply src/users/data/api.js:47]: did you verify this on local?

Comment (Ali-D-Akbar) [reply src/users/data/api.js:47]: Yes.

**Review by DawoudSheraz — APPROVED** [review]

**Review by azanbinzahid — APPROVED** [review]

**Pull request merged**